### DISCUSSION TITLE
Plans for more supported languages/framework in the future?

### DISCUSSION THREAD

**Discussion by ISnackable** [github/codeql#9508] [Q&A] [answered]
Just wondering if there are any plans for other languages?

**Comment by ISnackable** [comment, marked as answer]
Oh, seem like GitHub public roadmap: https://github.com/orgs/github/projects/4247/views/1?filterQuery=label%3Acodeql, does list some future plans. From what I can see, Koltin and Swift are among the one to be supported in the future. Is it possible to request for other languages, such as Rust?

Edit: Turns out there's a similar discussion. Here, https://github.com/github/codeql/discussions/8141. Closing this as duplicated.

**Reply by hmakholm** [reply]
Yes, the public roadmap contains the answer to the languages question we can share at the moment.

The roadmap doesn't show supports for particular frameworks. We're continually evaluating which under-supported frameworks it makes sense for us to spend engineering resources on writing support for -- but since the framework support is "just" a matter of QL code, we accept external contributions in this space too. There's also some work underway to make it easier to write and distribute external CodeQL support packs for frameworks in already supported languages -- stay tuned!

As the discussion on #8141 indicates, it is in principle possible to write external "database extractors" for CodeQL to create support for entirely new languages. The documentation we provide for _how_ to do that is admittedly extremely sparse and scattered at the moment; we intend to improve that eventually, but it has to compete with other demands for engineering time, so I can't offer a timeline.
